pages/admin/invite/userAccountPage/[id].tsx

```diff
@@ -6,13 +6,256 @@ import Combobox from "components/Combobox";
+              error={errors.email?.message}
+            >
+              {user?.email ? (
+                <input
+                  type="text"
+                  className="input input-full"
+                  name="email"
+                  defaultValue={user?.email}
+                  placeholder="e.g., soccer@fifa.com"
+                  ref={register}
+                />
+              ) : (
+                <input
+                  type="text"
+                  className="input"
+                  name="email"
+                  placeholder="e.g., soccer@fifa.com"
+                  ref={register}
+                />
+              )}
+            </FormField>
+            <FormField
+              label="Phone Number"
+              name="phoneNumber"
+              error={errors.phoneNumber?.message}
+            >
+              {user?.phoneNumber ? (
+                <input
+                  type="text"
+                  className="input"
+                  name="phoneNumber"
+                  defaultValue={user?.phoneNumber}
+                  placeholder="e.g., 123-456-7890"
+                  ref={register}
+                />
+              ) : (
+                <input
+                  type="text"
+                  className="input"
+                  name="phoneNumber"
+                  placeholder="e.g., 123-456-7890"
+                  ref={register}
+                />
+              )}
+            </FormField>
+            <FormField label="Role" name="role" error={errors.role?.message}>
+              {Object.values(UserRoleType).map((role: UserRoleType) => (
+                <label className="block font-normal" htmlFor={role} key={role}>
+                  <input
+                    className="mr-3"
+                    type="radio"
+                    name="role"
+                    id={role}
+                    defaultValue={role}
+                    onChange={() => setCurrRole(role)}
+                    checked={currRole === role}
+                    ref={register}
+                  />
+                  {UserRoleLabel[role]}
+                </label>
+              ))}
+            </FormField>
+          </fieldset>
+          <hr />
+          <div className="my-10">
+            <div className="mb-2 flex">
+              <Button className="button-primary px-10 py-2 mr-5" type="submit">
+                Save
+              </Button>
+              <Button
+                className="button-hollow px-10 py-2"
+                onClick={() => {
+                  setIsEditing(false);
+                }}
+              >
+                Cancel
+              </Button>
+            </div>
+            {error && <p className="text-red-600 text-sm">{error}</p>}
+          </div>
+        </form>
+      </div>
+    </Modal>
+  );
+};
+
+const UserAccountPage: React.FunctionComponent<UserRequest> = () => {
   const router = useRouter();
   const [selectedPlayers, setSelectedPlayers] = useState<User[]>([]);
```

Comment:
you have selectedPlayers here for the combo box but actually never use it, so in you accept API call you will actually pass in the status change as well as the new additional linked players if any. Linked players are passed in through as a role, so it will again look similar to what Shannon did in `[role]/id`
`const linkedPlayers: ViewingPermissionDTO[] = [];
    (relatedPlayers || [])?.forEach((role) => {
      const body = (JSON.stringify({
        type: currRole,
        userId: user?.id,
        relatedPlayerId: role.id,
      }) as unknown) as ViewingPermissionDTO;
      linkedPlayers.push(body);
    });`